kliff/trainer/lightning_trainer.py

```diff
@@ -0,0 +1,514 @@
+            to the model, therefore order, and the name of the arguments should be the
+            same as in the model definition. Example: ["x", "coords", "edge_index0", "edge_index1" ...,"batch"]
+        ckpt_dir: Directory to save the checkpoints
+        device: Device to run the model on. Default is "cpu"
+        ema: Whether to use Exponential Moving Average. Default is True
+        ema_decay: Decay rate for Exponential Moving Average. Default is 0.99
+        optimizer_name: Name of the optimizer to use. Default is "Adam"
+        lr: Learning rate for the optimizer. Default is 0.001
+        energy_weight: Weight for the energy loss. Default is 1.0
+        forces_weight: Weight for the forces loss. Default is 1.0
+        lr_scheduler: Name of the learning rate scheduler. Default is None
+        lr_scheduler_args: Arguments for the learning rate scheduler. Default is None
+    """
+
+    def __init__(
+        self,
+        model,
+        input_args: List,
+        ckpt_dir=None,
+        device="cpu",
+        ema=True,
+        ema_decay=0.99,
+        optimizer_name="Adam",
+        lr=0.001,
+        energy_weight=1.0,
+        forces_weight=1.0,
+        lr_scheduler=None,
+        lr_scheduler_args=None,
+    ):
+
+        super().__init__()
+        self.model = model
+        self.input_args = input_args
+        self.ckpt_dir = ckpt_dir
+        self.ema = ema
+        self.optimizer_name = optimizer_name
+        self.lr = lr
+        self.energy_weight = energy_weight
+        self.forces_weight = forces_weight
+
+        if is_torch_ema_present and ema:
+            ema = ExponentialMovingAverage(self.model.parameters(), decay=ema_decay)
+            ema.to(device)
+            self.ema = ema
+
+        self.lr_scheduler = lr_scheduler
+        self.lr_scheduler_args = lr_scheduler_args
+
+    def forward(self, batch: Any) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        Forward pass of the model. Returns the energy, and forces predicted by the model.
+        Args:
+            batch: dict containing torch tensors. It should at least have same keys as
+                defined in `self.input_keys`
+
+        Returns:
+            Energy and forces of the supplied batch. Please note that the forces are
+            partial forces, and hence needs to be summed to the contributing particles.
+
+        """
+        batch["coords"].requires_grad_(True)
+        model_inputs = {k: batch[k] for k in self.input_args}
+        predicted_energy = self.model(**model_inputs)
+        (predicted_forces,) = torch.autograd.grad(
+            [predicted_energy],
+            batch["coords"],
+            create_graph=True,  # TODO: grad against arbitrary param name
+            retain_graph=True,
+            grad_outputs=torch.ones_like(predicted_energy),
+        )
+        predicted_forces = scatter_add(predicted_forces, batch["images"], dim=0)
+        return predicted_energy, -predicted_forces
```

Comment:
Lets move the negative sign to one of the above line where `predicted_forces` is computed?